### ISSUE TITLE
Search Regex Error

### ISSUE THREAD

**Issue by kometchtech** [johngodley/search-regex#9]
I also get the following error.

```
Plugin: 2.0
WordPress: 5.4.1 (single)
PHP: 7.3.14
Browser: Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:78.0) Gecko/20100101 Firefox/78.0
JavaScript: https://blog.komeho.info/wp-content/plugins/search-regex/search-regex.js
REST API: https://blog.komeho.info/wp-json/

Error: パラメーター不足: replacePhrase (rest_missing_callback_param)
Action: search-regex/v1/replace/ POST
Params: {"searchPhrase":"http://www.amazon.co.jp","searchFlags":["case"],"source":["post"],"sourceFlags":[],"replacement":"https://www.amazon.co.jp","perPage":25,"page":50}
Code: 400 Bad Request
Raw: {"code":"rest_missing_callback_param","message":"\u30d1\u30e9\u30e1\u30fc\u30bf\u30fc\u4e0d\u8db3: replacePhrase","data":{"status":400,"params":["replacePhrase"]}}
```

**Comment by johngodley** [comment]
Can you try 2.0.1 and let me know if it solves the problem?

https://github.com/johngodley/search-regex/releases/tag/2.0.1

**Comment by kometchtech** [comment]
Thanks for the response.

I installed 2.0.1 directly and tried it, but it seems that work progress is not updated at all as shown in the following image. We waited at least 30 minutes and the situation didn't change.

![1589155434487](https://user-images.githubusercontent.com/5137077/81513926-f4ba0380-9366-11ea-9718-7c71b71fd5e1.jpg)

However, when I canceled and checked if the replacement was done, the replacement itself seems to have been done.

**Comment by johngodley** [comment]
Using the same search and replace I wasn't able to reproduce that problem. Can you open your browser's developer console and in the Network tab see what requests are being sent? If you can copy the response of one and send it to me then I'll be able to take a look.

**Comment by kometchtech** [comment]
Thanks for the call.
I tried to get a screenshot of the situation in the browser's developer console.

![search-regex](https://user-images.githubusercontent.com/5137077/81530244-73c92f00-939b-11ea-97ca-921cdfdb2820.png)

- Windows 10 Pro 1909
- Firefox 76.0.1

**Comment by johngodley** [comment]
Could you right click on the `/replace` request and then pick 'copy response` from the menu?

I have someone else with the same problem you reported saying that the new version fixes it for them, so I'm not sure why your progress bar isn't moving.

**Comment by kometchtech** [comment]
Thanks for the quick response.
I got a copy and I'm going to paste it.
![無題](https://user-images.githubusercontent.com/5137077/81531270-5eed9b00-939d-11ea-86df-2acf35933f98.png)

```
{"results":{"rows":3,"phrases":3},"progress":{"current":0,"rows":0,"previous":false,"next":50},"totals":{"rows":4628,"matched_rows":0,"matched_phrases":0}}
```
```
{"wp-auth-check":true,"server_time":1589178892}
```

**Comment by johngodley** [comment]
Is that the only request that is made to `/wp-json/search-regex`? If you leave it a bit are any other requests made?

**Comment by kometchtech** [comment]
Here's what happened on the browser side, which I left alone until I heard back from you.

![無題](https://user-images.githubusercontent.com/5137077/81532532-834a7700-939f-11ea-9caa-45f32411261e.png)

**Comment by johngodley** [comment]
Ok, thanks for all the details. It does look like it's stopped. I'll see if I can reproduce it using this information.

**Comment by kometchtech** [comment]
Please let me know if there is anything else I can do to help.

**Comment by souzouno-yakata** [comment]
I am also seeing the same problem
Some time ago it was working fine

**Comment by souzouno-yakata** [comment]
We operate at multi-site
Some sites seem to work fine.

**Comment by johngodley** [comment]
@souzouno-yakata do you know what the difference is between the working and non-working sites?

**Comment by souzouno-yakata** [comment]
About the difference,I do not understand at all.
Both are subsites, Common database.
Just because the progress bar doesn't move,
Processing is finish.

**Comment by johngodley** [comment]
Can you try the 2.1 beta here and let me know if it helps?

https://github.com/johngodley/search-regex/releases/tag/2.1

**Comment by souzouno-yakata** [comment]
I installed two beta versions and compared them. There was no difference in the problem.

But there is something new.

The progress bar does not update if it contains a double byte "?".

An example
フィットハイブリッドが燃費でアクアに勝てないのはなぜか？
（Why is Fit Hybrid unable to beat Aqua due to fuel consumption?）

**Comment by johngodley** [comment]
> The progress bar does not update if it contains a double byte "?".

If what contains a ? - the search or the replace?

This problem only seems to affect Japanese sites.

**Comment by souzouno-yakata** [comment]
It is Search.

Replace has no problem

**Comment by philip47** [comment]
The old version 1.4.16 is working fine without issues, the new version is ugly and useless.
How to downgrade: https://kinsta.com/knowledgebase/download-older-versions-of-wordpress-plugins/
I hope it helps you all.

**Comment by johngodley** [comment]
Phillip, I've deleted your comment. If you can't say things in a more constructive way then you are not welcome here.

**Comment by johngodley** [comment]
Alright, I think I've reproduced the problem and it should be fixed in 2.1 beta 3:

https://github.com/johngodley/search-regex/releases/tag/2.1

Let me know how it goes.

**Comment by souzouno-yakata** [comment]
Thanks for the fix.
It looks fine.
Japanese locale is also nice.

**Comment by kometchtech** [comment]
It takes about 5 minutes to get it to finish normally.
Thank you for the fix.

By the way, is it the intended behavior that the maximum number of replacements is 65?

**Comment by johngodley** [comment]
No, there is no limit.

**Comment by kometchtech** [comment]
As for the behavior at my time, I did a substitution when 110 cases were targeted, but for some reason only 65 cases seemed to be substituted then.

**Comment by souzouno-yakata** [comment]
Hi kometchtech,

Are you hitting the search button first and then the replace button?

The search string may change when the search button is pressed first. You don't notice this.
This is probably due to the browser.

How to improve
1. Press the search button and then re-enter the first search string.
2. The plug-in does the same operation

**Comment by kometchtech** [comment]
Hi @souzouno-yakata ,

Thanks for the information.

> Are you hitting the search button first and then the replace button?
> 
> The search string may change when the search button is pressed first. You don't notice this.
> This is probably due to the browser.

Yes. I was making it so that after the search confirmed the string to be searched for (after pressing the search button), it would replace it (like pressing the replace button).

**Comment by souzouno-yakata** [comment]
Stop pressing the search button first,
If you press the replace button first.
I think it will be done as you want.

**Comment by kometchtech** [comment]
Thanks info.

> Stop pressing the search button first,
> If you press the replace button first.
> I think it will be done as you want.

I think I understand the procedure you are describing.
However, I don't have the replacement data at the moment, so I'll try it when the replacement work occurs.

**Comment by johngodley** [comment]
This should all now be fixed with version 2.1. If there are any other problems please open them in a new issue, this is getting quite long now!
